### ISSUE TITLE
Aurelia WordPress

### ISSUE THREAD

**Issue by adriatic** [Vheissu/aurelia-graphql-apollo#1]
Hello @Vheissu - we lost contact for a few years, but it seems that a set of common interest might bridge the gap again. At this point, I would like to understand the reasons and depths of your interest in aurelia-graphql. In my recent chat with Rob, he showed interest in Aurelia WordPress integration and I am tickled by the **[The Story Behind the New WordPress.com](https://developer.wordpress.com/2015/11/23/the-story-behind-the-new-wordpress-com/)** and **[wp-calypso](https://github.com/Automattic/wp-calypso)** in particular.

This, about year old interest was created by my desire to finally after 50 years do something for myself, so I started to learn WordPress as likely most potent CMS framework. I liked what I saw - but was very hesitant to learn PHP. 

In my opinion qraphql-apollo is the obvious candidate to create aurelia wordpress integration - and Rob mentioned that you are doing something there as well. He will call me back after interviewing you, to tell me how much he cares about such project, so I might contact you again.

**To summarize**: what is the status of your aurelia-graphql-apollo project?

**Comment by adriatic** [comment]
P.S. Just realized that apollo-boost was indeed integrated on the "main product", allowing you to finish this project.

![image](https://user-images.githubusercontent.com/2712405/60682590-d1c6e500-9e61-11e9-949a-0a767d4a393a.png)

**Comment by Vheissu** [comment]
@adriatic It's really good to hear from you. It's nice to see you're still active in the Aurelia community.

To me, Wordpress has become more like a fully-featured API with its integrated REST API than it has a CMS. When Wordpress integrated that API functionality out-of-the-box, they made it so you could run Wordpress headless and use whatever client you wanted and have that really nice Wordpress admin panel that you get for managing content and its trove of plugins.

- In terms of adding GraphQL to Wordpress, I highly recommend this plugin: https://www.wpgraphql.com/
- I actually started exploring integrating Aurelia into a Wordpress thing quite a while ago in this repo here: https://github.com/Vheissu/aureliapress (it doesn't use GraphQL though).

I feel like the next logical step would be integrating Aurelia into a Wordpress theme and adding in first-class support for GraphQL. To me, GraphQL is the greatest thing to happen to API's in a very long time,. the ability to query for only what you need and nothing more really appeals to me.

First things first, I'm going to update this repo :)

**Comment by adriatic** [comment]
Thanks for such warm and quick response, Dwayne 😄 .

Comment on your first sentence:
> To me, Wordpress has become more like a fully-featured API with its integrated REST API than it has a CMS

I'll quote  **[wikipedia](https://en.wikipedia.org/wiki/Content_management_system)** as my own opinion:
> A content management system (CMS) manages the creation and modification of digital content. It typically supports multiple users in a collaborative environment..

This AFAIK is a precise definition of WordPress - with the caveat that I do not know WordPress sufficiently to make any authoritative statements. I decided to learn WordPress to the extent necessary to create a relatively ambitious WordPress app used to organize last 40 years of pictures, articles, videos and music related to my (pretty distributed) family (see my then 28 years old spouse as page taken from a blog maintained in this app)

![image](https://user-images.githubusercontent.com/2712405/60694874-b4b80380-9ead-11e9-9d3a-5c0c8cc3b605.png)

As I worked on this app, I realized that WordPress "father" (Matt Mullenweg) has stated all my accumulated objections in an article written 5 years ago, describing how would he create WordPress if he had today's tools, then. Nevertheless, I god so excited about technologies, like GraphQL, Prisma, etc (see https://medium.com/@xoor/coding-a-graphql-api-with-node-js-c02d617f49f4 - one of the first articles I run into)

My plan is to create a sufficiently attractive "model" of the real Aurelia WordPress framework to test Rob's emotions about it (I dd tell him that, so he mentioned that he will try to learn from you what is the landscape in this domain). This model is going to be different from https://github.com/Automattic/wp-calypso - they went way overboard trying to cook all the ingredients at once - and they also want to use REST API, which in my "humble" opinion is similar to build a steam train at the time when "mag-lev" technology is available. I tried to just build and run wp-calypso and failed on too many steps (look at 35691 commits, 699 branches and 1594 open issues and you will see that the team is composed from PHP developers, learning JavaScript while working on this project.

Since in my attempts to talk you into "doing something together" I spent 10 times the amount of energy that would be considered "polite", I am only planning to keep you informed about what I do, since it seems obvious that you care about this general context.
